src/commands/init.ts

```diff
@@ -28,20 +29,87 @@ export class InitCommand implements Command {
 
   run(options, config): Promise<any> {
     // Defer dependency loading until this specific command is run
-    const PolykartGenerator = require('../templates/polykart').PolykartGenerator;
+    const PolykartGenerator =
+        require('../templates/polykart').PolykartGenerator;
+    const findup = require('findup');
+    const inquirer = require('inquirer');
     const YeomanEnvironment = require('yeoman-environment');
 
     return new Promise((resolve, reject) => {
       let logger = logging.getLogger('cli.init');
-      let templateName = options['name'] || 'polymer-init';
 
       logger.debug('creating yeoman environment...');
       let env = new YeomanEnvironment();
-      env.register(require.resolve('generator-polymer-init'), 'polymer-init:app');
-      env.registerStub(PolykartGenerator, 'polykart:app');
+      env.register(
+          require.resolve('generator-polymer-init'), 'polymer-init-basic:app');
+      env.registerStub(PolykartGenerator, 'polymer-init-shop:app');
+
+      env.lookup(() => {
+        let generators = env.getGeneratorsMeta();
+
+        let runGenerator = (generatorName: string, templateName?: string) => {
```

Comment:
I basically default to arrow functions for everything exception functions which need to be able to have `this bound. That way I don't accidentally forget to `.bind()`, and the presence of `function` means I should pay attention because `this` is going to be interesting.